includes/Modules/AdSense/AMP_Tag.php

```diff
@@ -78,9 +78,26 @@ protected function render() {
 
 		$this->adsense_tag_printed = true;
 
+		$amp_auto_ads_opt = array(
+			'ad-client' => $this->tag_id,
+		);
+
+		$amp_auto_ads_opt_filtered = apply_filters( 'googlesitekit_amp_auto_ads_attributes', $amp_auto_ads_opt, $this->tag_id );
+
+		if ( ! is_array( $amp_auto_ads_opt_filtered ) || empty( $amp_auto_ads_opt_filtered ) ) {
+			$amp_auto_ads_opt_filtered = $amp_auto_ads_opts;
+		}
+
+		$amp_auto_ads_opt_filtered['ad-client'] = $this->tag_id;
```

Comment:
Let's probably extract it into a separate function and then re-use it for both `render_story_auto_ads` and `render` methods:

```php
private function get_auto_ads_attributes() {
    $options = array(
        'ad-client' => $this->tag_id,
    );

    if ( ! empty( $this->story_ad_slot_id ) ) {
        $options['ad-slot'] = $this->story_ad_slot_id;
    }

    $filtered_options = apply_filters( 'googlesitekit_amp_auto_ads_attributes', $amp_auto_ads_opt, $this->tag_id, $this->story_ad_slot_id );
    if ( is_array( $filtered_options ) && ! empty( $filtered_options ) ) {
        $options = $filtered_options;
        $options['ad-client'] = $this->tag_id;
    }

    return $options;
}

public function render() {
    ...
    
    $attributes = '';
    for ( $this->get_auto_ads_attributes() as $key => value ) {
        $attributes .= sprintf( ' data-%s="%s"', esc_attr( $key ), esc_attr( $value ) );
    }

    ...
}

public function render_story_auto_ads() {
    ...
 
    $attributes = array();
    for ( $this->get_auto_ads_attributes() as $key => value ) {
        $attributes[ 'data-' . $key ] = $value;
    }

    ...
}
```

Reply:
Nice. I've added the filter name as parameter since the filter name is different.